bindings/azure/cosmosdb/cosmosdb.go

```diff
@@ -62,56 +59,57 @@ func (c *CosmosDB) Init(metadata bindings.Metadata) error {
 	c.partitionKey = m.PartitionKey
 
 	// Create the client; first, try authenticating with a master key, if present
-	var config *documentdb.Config
+	var client *azcosmos.Client
 	if m.MasterKey != "" {
-		config = documentdb.NewConfig(&documentdb.Key{
-			Key: m.MasterKey,
-		})
+		cred, keyErr := azcosmos.NewKeyCredential(m.MasterKey)
+		if keyErr != nil {
+			return keyErr
+		}
+		client, err = azcosmos.NewClientWithKey(m.URL, cred, nil)
+		if err != nil {
+			return err
+		}
 	} else {
 		// Fallback to using Azure AD
-		env, errB := azure.NewEnvironmentSettings("cosmosdb", metadata.Properties)
-		if errB != nil {
-			return errB
+		env, errEnv := azure.NewEnvironmentSettings("cosmosdb", metadata.Properties)
+		if errEnv != nil {
+			return errEnv
 		}
-		spt, errB := env.GetServicePrincipalToken()
-		if errB != nil {
-			return errB
+		token, errToken := env.GetTokenCredential()
+		if errToken != nil {
+			return errToken
 		}
-		config = documentdb.NewConfigWithServicePrincipal(spt)
-	}
-	// disable the identification hydrator (which autogenerates IDs if missing from the request)
-	// so we aren't forced to use a struct by the upstream SDK
-	// this allows us to provide the most flexibility in the request document sent to this binding
-	config.IdentificationHydrator = nil
-	config.WithAppIdentifier("dapr-" + logger.DaprVersion)
-
-	c.client = documentdb.New(m.URL, config)
-
-	// Retries initializing the client if a TooManyRequests error is encountered
-	err = retryOperation(func() (err error) {
```

Comment:
We still need a backoff retry during the init (checking for 429 rate limit exceeded) because CosmosDB on client init downloads a ton of metadata under the hood which quickly result in API rate limits being exceeded when starting up multiple Dapr instances simultaneously. This was actually a customer issue we fixed in the past.

Reply:
@ItalyPaleAle can you please comment on this since you told @RyanLettieri that the new SDK handles this.

Reply:
Per @elsur (who maintains that SDK):

>  The Azure Go SDKs (all) have a common retry policy, which includes 429 handling: https://github.com/Azure/azure-sdk-for-go/blob/b386adc15c2e60e3812956f68d244b608330c479/sdk/azcore/runtime/policy_retry.go#L48

Reply:
Per @ealsur (who maintains that SDK):

>  The Azure Go SDKs (all) have a common retry policy, which includes 429 handling: https://github.com/Azure/azure-sdk-for-go/blob/b386adc15c2e60e3812956f68d244b608330c479/sdk/azcore/runtime/policy_retry.go#L48

> Looks like it defaults to 3 retries but you can change it https://github.com/Azure/azure-sdk-for-go/blob/b386adc15c2e60e3812956f68d244b608330c479/sdk/azcore/runtime/policy_retry.go#L25

Reply:
If needed, we can (should?) change the [`policy.RetryOptions` object](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime#NewRetryPolicy)

Reply:
We absolutely should. Let me link you to the PR where we fixed this in the Cosmos DB state store which absolutely could be an issue here. https://github.com/dapr/components-contrib/pull/1329

I don't remember whether it was Zeiss or Bosch bringing up this concern, but either way - it was an important issue which we were only able to address with exponential backoff for the client initialization.

Reply:
It does do exponential backoff already. It's configured to try 3 times with an exponential backoff of up to 2 mins. We need to tweak it only if we don't think that's good/enough.